src/EditorFeatures/Core/Interactive/InteractiveWorkspace.cs

```diff
@@ -16,20 +16,17 @@ internal partial class InteractiveWorkspace : Workspace
         private SourceTextContainer? _openTextContainer;
         private DocumentId? _openDocumentId;
 
-        internal InteractiveWorkspace(HostServices hostServices, IGlobalOptionService globalOptions)
+        internal InteractiveWorkspace(HostServices hostServices)
             : base(hostServices, WorkspaceKind.Interactive)
         {
             // register work coordinator for this workspace
-            if (globalOptions.GetOption(SolutionCrawlerRegistrationService.EnableSolutionCrawler))
-            {
-                Services.GetRequiredService<ISolutionCrawlerRegistrationService>().Register(this);
-            }
+            DiagnosticProvider.Enable(this);
```

Comment:
consumers do not check anymore.  they just call into service to enable themselves. the service decides if it needs to do anything or not.